shared/profile/block/index.tsx

```diff
@@ -0,0 +1,64 @@
+import * as React from 'react'
+import * as Kb from '../../common-adapters'
+import * as Styles from '../../styles'
+import * as Constants from '../../constants/profile'
+
+type Props = {
+  username: string
+  errorMessage?: string | null
+  idle: boolean
+  onClose: () => void
+  onSubmit: () => void
+  isWaiting: boolean
+}
+
+class Block extends React.Component<Props> {
+  componentDidUpdate(prevProps: Props) {
+    if (!prevProps.idle && this.props.idle) {
+      this.props.onClose()
+    }
+  }
+
+  render() {
+    const props = this.props
+    let errorBox = null
+    if (props.errorMessage) {
+      errorBox = (
+        <Kb.Box style={styles.errorBanner}>
+          <Kb.Text center={!Styles.isMobile} style={styles.errorBannerText} type="BodySemibold">
+            {props.errorMessage}
+          </Kb.Text>
+        </Kb.Box>
+      )
+    }
+    const modalProps = {
+      confirmText: 'Yes, block them',
+      description:
+        'This will hide them from your followers and suggestions, and prevent them from creating new conversations or teams with you. Note that they may be able to find out that you block them.',
```

Comment:
Maybe "that you blocked them"?  Your call.

Reply:
I tried that. But 'them' wrapped onto its own line and looked silly. Which made @cecileboucheron look like a magician. What do you think?
(I'm merging but still listening)